### PR TITLE
add related objs

### PR DESCRIPTION
**Pull request by juanvallejo**
compiles - just need to test locally
Updates deps in order to bring in the new `.Status.RelatedObject` field from openshift/api

cc @soltysh @deads2k

### CODE REVIEW

**Comment by ravisantoshgudimetla** [comment]
@juanvallejo - Does this bring in version too?

**Comment by juanvallejo** [comment]
@ravisantoshgudimetla 
> Does this bring in version too?

It brings in `.Status.Versions` (operand versions)

**Comment by juanvallejo** [comment]
Currently debugging this locally. Status controller appears to be crash looping here https://github.com/openshift/library-go/blob/master/pkg/operator/status/status_controller.go#L82 and never brings up the clusteroperator resource

**Comment by ravisantoshgudimetla** [comment]
Great @juanvallejo - Thanks for this PR. I am wondering if we need https://github.com/openshift/cluster-kube-scheduler-operator/pull/50 if this PR is anyway bumping library-go.

/cc @joelsmith for visibility as he started looking into this.
/cc @sjenning

**Comment by ravisantoshgudimetla** [comment]
> Currently debugging this locally. Status controller appears to be crash looping here https://github.com/openshift/library-go/blob/master/pkg/operator/status/status_controller.go#L82 and never brings up the clusteroperator resource

I have started adding making some changes on the same lines for adding a configmap in `GlobalUserSpecifiedConfigNamespace` but I can copy most of my changes from this PR if this passes. So, I will try this locally.

**Comment by juanvallejo** [comment]
>  I am wondering if we need #50 if this PR is anyway bumping library-go.

Main goal behind this one is to bump openshift/api. Side effect does end up being that it brings in an even newer library-go.

**Comment by juanvallejo** [comment]
I've opened https://github.com/openshift/cluster-kube-scheduler-operator/pull/55 to test this PR. It bumps library go to [commit https://github.com/openshift/library-go/commit/7043c4ecdf45532bf6237c886a1544b01fc46fdc](https://github.com/openshift/library-go/commit/7043c4ecdf45532bf6237c886a1544b01fc46fdc) which still adds support for .Status.RelatedObjects, without bringing in the most recent changes in `library-go/master`.

https://github.com/openshift/cluster-kube-scheduler-operator/pull/55 actually appears to be working fine. It adds the `.RelatedObjects` to the clusteroperator status, as expected, and no crash looping is observed

**Comment by juanvallejo** [comment]
Quick update: was able to debug the issue. Updated this PR accordingly. It should be working as expected now.

**Comment by juanvallejo** [comment]
Quick update: was able to debug the issue. Updated this PR accordingly. It should be working as expected now.

File: pkg/operator/resourcesynccontroller/resourcesynccontroller.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	kubeClient kubernetes.Interface,
+	eventRecorder events.Recorder) (*resourcesynccontroller.ResourceSyncController, error) {
+
+	resourceSyncController := resourcesynccontroller.NewResourceSyncController(
+		operatorConfigClient,
+		kubeInformersForNamespaces,
+		kubeClient,
+		eventRecorder,
+	)
+	if err := resourceSyncController.SyncConfigMap(
```
Comment (juanvallejo) [inline pkg/operator/resourcesynccontroller/resourcesynccontroller.go]: @ravisantoshgudimetla I'm assuming this is the wrong configmap for this operator. Are there any resources that this operator will need to sync?

Comment (juanvallejo) [reply pkg/operator/resourcesynccontroller/resourcesynccontroller.go]: cc @deads2k ^

Comment (ravisantoshgudimetla) [reply pkg/operator/resourcesynccontroller/resourcesynccontroller.go]: At this point of time, no, I am planning to add one soon.

Comment (juanvallejo) [reply pkg/operator/resourcesynccontroller/resourcesynccontroller.go]: Okay. Will remove this block from the PR then. Will still leave the empty resourcesynccontroller though, to at least make adding those easier in the future

**Comment by deads2k** [comment]
> @juanvallejo - Thanks for the PR, the latest changes are bringing in all the fields we need but I am curious, why versions are being shown as null.

The versions API updated to reflect the potential for 0, 1, or many versions.  You'll need to separately update that aspect.

**Comment by ravisantoshgudimetla** [comment]
Got it @deads2k. Thank you..

/hold cancel

**Pull request merged**